### PR TITLE
34 money managing tips

### LINKED ISSUE #34: As a user, I want to view recommended money-managing tips from the application
... so that I can better improve my money-managing habits

### LINKED ISSUE #39: As a forgetful user, I want to receive reminders on how I should spend my allowance
... so that I can be consciously aware of my budget constraints

### LINKED ISSUE #111: As a user, I want to set and update my budget
... so that I can limit my spending against a budget

### LINKED ISSUE #128: [Bug] Addition of income does not generate correct acknowledgement message
image

In this screenshot, the default monthly budget value is taken to be at default at $1000. When a new Income transaction is added with `add t/income c/salary a/2000 d/30092022 i/jan_salary`, the acknowledgement message shows a negative value, due to incorrect operation of the income values. Supposedly, the income should be incremented to the monthly budget.

### PR DESCRIPTION
**Pull request by wcwy**
The application will now check if the monthly budget has been exceeded on every add, edit and delete command and to display the budget and tips to the users. Furthermore, the application will also display the budget remained/exceeded for the current month when the application is started. 

This PR aims to enhance on the budget feature #111 to display useful messages related to budgets.
It also aims to work on recommending money-managing tips #34 by displaying tips to user on every add, delete or edit transaction on whether the user should consider spending less or simply keep up the good spending habits.
It also aims to complete reminder #39 by displaying reminders on whether the budget has been exceeded for the current month during the start of program.

This PR also fixes the bug identified in #128.

### CODE REVIEW

File: src/main/java/seedu/duke/common/InfoMessages.java [hunk 1]
```diff
@@ -38,7 +44,12 @@ public enum InfoMessages {
     INFO_PURGE_WARNING("Are you sure you want to proceed with this command? Please enter 'Y' to confirm."),
     INFO_BUDGET_SET_SUCCESSFUL("You have successfully updated the budget."),
     INFO_CURRENT_BUDGET("Monthly budget set as: $"),
-    INFO_REMAINING_BUDGET("Budget remained for the month of transaction: $");
+    INFO_REMAINING_BUDGET("Budget remained for "),
+    INFO_EXCEEDING_BUDGET("Budget exceeded for "),
+    INFO_BUDGET_EXCEEDED_TIPS("Consider spending less!"),
+    INFO_BUDGET_NOT_EXCEEDED_TIPS("Keep it up!"),
```
Comment (chydarren) [inline src/main/java/seedu/duke/common/InfoMessages.java:50]: Clear naming of `enum` variables!

File: src/main/java/seedu/duke/data/Budget.java [hunk 2]
```diff
@@ -53,6 +108,67 @@ public static String getBudgetLeft(long totalMonthlyExpense) {
...
+     * Thus, the budget is exceeded and returns true.
+     *
+     * <p>If the budget is not positive, it means that the total monthly expense is lower than or equals to the
+     * budget left. Thus budget is not yet exceeded and returns false.
+     *
+     * @param budgetLeft A long value indicating the difference of total monthly expense and monthly budget.
+     * @return A boolean value indicating whether the budget has been exceeded for the month.
+     */
+    private static boolean hasExceededBudget(long budgetLeft) {
+        return budgetLeft < 0;
```
Comment (chydarren) [inline src/main/java/seedu/duke/data/Budget.java:127]: Consider changing the `0` to a magic number, maybe like `PROHIBITED_BUDGET` or something for more meaning.

File: src/main/java/seedu/duke/data/TransactionList.java [hunk 3]
```diff
@@ -394,7 +394,7 @@ public String listTimeStats(ArrayList<Transaction> timeTransactions, int year, i
         // Formats every entry in the hashmap into a categorical savings list
         for (Transaction entry : timeTransactions) {
             timeSavingsList += String.format("%s%s%s %s%s%s", PREFIX_CATEGORY, entry.getCategory(),
-                    POSTFIX_CATEGORY, SYMBOL_DOLLAR, entry.getAmount(), LINE_SEPARATOR);
+                    POSTFIX_CATEGORY, DOLLAR_SIGN, entry.getAmount(), LINE_SEPARATOR);
```
Comment (chydarren) [inline src/main/java/seedu/duke/data/TransactionList.java:397]: Good attempt at renaming the dollar sign!

**Review by chydarren — APPROVED** [review]
Great attempt at adding some money managing tips feature to complete the Moolah Manager Application for V2.0! Thanks for changing some of the variable names and implementing adequate J-Unit tests for your portion too. The format of the J-Unit tests serve as a good example for the rest of the team when they implement theirs.

**Comment by wcwy** [comment]
Fixes #39

**Pull request merged**